File: homework_iv/homework_iv_database/src/main/java/homework_iv/database/service/DefaultRestResponseService.java
package homework_iv.database.service;

import homework_iv.database.model.Assistant;
import homework_iv.database.repository.AssistantRepository;
import lombok.SneakyThrows;
import org.json.JSONObject;
import org.json.JSONTokener;

import java.io.BufferedReader;
import java.io.PrintWriter;

public class DefaultRestResponseService implements RestResponseService {

    private final AssistantRepository assistantRepository;

    public DefaultRestResponseService(AssistantRepository assistantRepository) {
        this.assistantRepository = assistantRepository;
    }

    @Override
    @SneakyThrows
    public void saveAssistant(BufferedReader socketIn, PrintWriter socketOut) {
        JSONTokener token = new JSONTokener(socketIn.readLine());
        JSONObject request = new JSONObject(token);

        Assistant newEntry = Assistant.builder()
                .name(request.getString("name"))
                .score(request.getInt("score"))
                .build();

        assistantRepository.saveNewAssistant(newEntry);

        socketOut.println("saved");
    }

    @Override
    public void listAssistants(PrintWriter socketOut) {
        JSONObject response = new JSONObject();

        assistantRepository.getAssistantAverage().forEach(assistant -> {
            JSONObject assistantObject = new JSONObject();
            assistantObject.put("name", assistant.getName());
            assistantObject.put("averageScore", assistant.getAverageScore());
            response.append("assistants", assistantObject);
        });

        socketOut.println(response.toString());
        socketOut.println("listed");
    }
}

File: homework_vi/src/main/java/rs/webprogramming/homework_vi/service/DefaultRestResponseService.kt
package rs.webprogramming.homework_vi.service

import rs.webprogramming.homework_vi.model.Assistant
import org.json.JSONObject
import rs.webprogramming.homework_vi.repository.AssistantRepository
import rs.webprogramming.homework_vi.repository.MySqlAssistantRepository
import javax.servlet.http.HttpServletRequest
import javax.servlet.http.HttpServletResponse

class DefaultRestResponseService: RestResponseService {

    private val assistantRepository: AssistantRepository = MySqlAssistantRepository()

    override fun invalidScore(response: HttpServletResponse, assistant: Assistant) {
        val jsonResponse = JSONObject()

        jsonResponse.put("httpResponse", 400)
        jsonResponse.put("successful", false)
        jsonResponse.put("errorMessage", "Invalid score: [${assistant.score}] - score must be in range [1, 10]")

        response.writer.println(jsonResponse.toString())
        response.contentType = "application/json"
        response.status = 400
    }

    override fun saveAssistant(request: HttpServletRequest, response: HttpServletResponse, assistant: Assistant) {
        assistantRepository.saveNewAssistant(assistant)

        val jsonResponse = JSONObject()

        jsonResponse.put("httpResponse", 200)
        jsonResponse.put("successful", true)

        response.writer.println(jsonResponse.toString())
        response.contentType = "application/json"
        response.status = 200

        val requestDispatcher = request.getRequestDispatcher("/assistants.jsp")
        requestDispatcher.forward(request, response)
    }

}